traffic_ops/traffic_ops_golang/server/servers.go

```diff
@@ -223,9 +223,12 @@ func getServers(params map[string]string, tx *sqlx.Tx, user *auth.CurrentUser) (
 FULL OUTER JOIN deliveryservice_server dss ON dss.server = s.id
 `
 		// depending on ds type, also need to add mids
-		dsType, err := deliveryservice.GetDeliveryServiceType(dsID, tx.Tx)
+		dsType, exists, err := deliveryservice.GetDeliveryServiceType(dsID, tx.Tx)
 		if err != nil {
-			return nil, err, nil, http.StatusBadRequest
+			return nil, nil, err, http.StatusInternalServerError
+		}
+		if !exists {
+			return nil, errors.New("a deliveryservice with id '" + strconv.Itoa(dsID) + "' was not found"), nil, http.StatusBadRequest
```

Comment:
nit: `fmt.Errorf` is more clean for formatting errors than concatenation of string values in `errors.New`.